airflow/api_connexion/security.py

```diff
@@ -48,10 +61,194 @@ def requires_access_decorator(func: T):
         @wraps(func)
         def decorated(*args, **kwargs):
             check_authentication()
-            if appbuilder.sm.check_authorization(permissions, kwargs.get("dag_id")):
+            if appbuilder.sm.check_authorization(permissions):
```

Comment:
1. Should we raise deprecation warning if the decorator is used by someone?

2. Should we raise exception here if auth_manager is not FAB in case we run a function decorated with it? 

This would give people indication that if they want to use their own plugins, they should switch to the other methods. 

We could also make a copy of that decorator in FAB auth manager and make sure that all the remaining Airflow embeded APIs (user/roles) do not raise deprecation warnings. Those are just a few lines to copy but this way only user's code would generate deprecation warnings.

Reply:
I was thinking removing this one once the user and roles endpoints have been migrated but I guess it is the same case as `has_access` decorator and we should keep it. I will:
- Raise a deprecation warning when used
- Copy it to the FAB auth manager and call this one from the one above 

> Should we raise exception here if auth_manager is not FAB in case we run a function decorated with it?

I really want to avoid doing `if not isistance(auth_manager, FabAuthManager` as much as possible. I think deprecation warnings are enough.

Reply:
Done